java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/TransportPort.java

```diff
@@ -20,7 +21,7 @@
     private final static int NONE_VAL = 0;
     public final static TransportPort NONE = new TransportPort(NONE_VAL);
 
-    public static final TransportPort NO_MASK = new TransportPort(0xFFFFFFFF);
+    public static final TransportPort NO_MASK = new TransportPort(MAX_PORT);
```

Comment:
bugfix here

Reply:
@Sovietaced a bit more context: for VFT internal validation logic, it is important that a flow that we have sent to the switch comes back in exactly the same way we've sent it. 

This means that for cases where the switch applies normalization to the flow entry, the controller must do the same thing. In this case, there is an issue with the normalization for tcp port entries that have a mask: Whenever the controller sends a tcp masked entry with a mask of `0xFFFF`, i.e., all bits set, the switch treats that as an unmasked TCP port entry (which make sense, because the mask just matches the whole value).

The controller attempts to do the same normalization here, in `OFOxmTcpDstMasked`:
https://github.com/floodlight/loxigen-artifacts/blob/master/openflowj/gen-src/main/java/org/projectfloodlight/openflow/protocol/ver14/OFOxmTcpDstMaskedVer14.java#L92

It compares the value of the mask to `NO_MASK` and then returns an unmasked value if it's the same. However, the value of the constant here is wrong: even tho the port is only a two-byte value (so the mask can really only be `0xFFFF`); the value defined here is `0xFFFFFFFF`, so the mask for `tcp_dst` (and similarly for `tcp_src` and `udp_src/dst`) will never be dropped.

Reply:
Thanks for the explanation. That makes more sense now.